test/intl402/Temporal/Duration/compare/relativeto-hour.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.duration.compare
+description: relativeTo with hours.
+features: [Temporal]
+---*/
+
+const oneDay = new Temporal.Duration(0, 0, 0, 1);
+const hours24 = new Temporal.Duration(0, 0, 0, 0, 24);
+
+assert.sameValue(
+  Temporal.Duration.compare(oneDay, hours24, { relativeTo: Temporal.ZonedDateTime.from('2017-01-01T00:00[America/Montevideo]') }),
+  0,
+  'relativeTo does not affect days if ZonedDateTime, and duration encompasses no DST change');
+assert.sameValue(
+  Temporal.Duration.compare(oneDay, hours24, { relativeTo: Temporal.ZonedDateTime.from('2019-11-03T00:00[America/Vancouver]') }),
+  1,
+  'relativeTo does affect days if ZonedDateTime, and duration encompasses DST change');
+assert.sameValue(
+  Temporal.Duration.compare(oneDay, hours24, { relativeTo: '2019-11-03T00:00[America/Vancouver]' }),
+  1,
+  'casts relativeTo to ZonedDateTime from string');
+assert.sameValue(
+  Temporal.Duration.compare(oneDay, hours24, {
+    relativeTo: { year: 2019, month: 11, day: 3, timeZone: 'America/Vancouver' }
+  }),
+  1,
+  'casts relativeTo to ZonedDateTime from object');
+
```

Comment:
I think we need to have these tests in the `built-ins/` suite as well. I would suggest using `TemporalHelpers.springForwardFallBackTimeZone()` to create a time zone that has a DST change, in a way that hosts without 402 can execute. For the first assertion where the duration encompasses no DST change, you could just use UTC instead of America/Montevideo.

Reply:
Are the tests in [twenty-five-hour-day.js](https://github.com/tc39/test262/blob/28b31c0bf1960878abb36ab8597a0cae224a684d/test/built-ins/Temporal/Duration/compare/twenty-five-hour-day.js) sufficient?

Reply:
They do look like they cover somewhat the same thing. I don't know if they cover the case where we get a different answer out of Duration.compare() depending on whether the string or property bag value for `relativeTo` converts to a PlainDate or ZonedDateTime.

Reply:
I think you'll be able to write that test more quickly than me :)

Reply:
Fair enough, I'll include it in my next PR.